### PR TITLE
Check V6 Processing for Thyroid Profile #21

### PR DESCRIPTION
**Pull request by Ceibo**
First, we want to ensure that we’re correctly handling V6 for Thyroid Profile when it’s fetched for the first time and there are no final results.

Then, we want to confirm that the second time we receive those results, when they are in **PARTIAL** status, with some items **PENDING** and others **COMPLETED**, everything is processed correctly.

### CODE REVIEW

File: src/providers/antechV6-mapper.ts [hunk 1]
```diff
@@ -262,6 +260,10 @@ export class AntechV6Mapper {
       return ResultStatus.REVISED
     }
 
+    if ((result.PendingTestCount ?? 0) > 0 && (result.PendingTestCount ?? 0) < (result.TotalTestCount ?? 0)) {
+      return ResultStatus.PARTIAL;
+    }
+
```
Comment (monsieurBelbo) [inline src/providers/antechV6-mapper.ts:266]: The redundant null coalescing operators makes this conditional a bit confusing. Consider simplifying it, maybe by extracting the pending count to another variable.

Also, if we can't completely count on the counts provided by their API, maybe we should consider inspecting the status of the `UnitCodeResults`.

Consider making `extractResultStatus` public so it can be tested for all these scenarios.

File: src/providers/antechV6-mapper.ts [hunk 2]
```diff
@@ -187,6 +190,26 @@ export class AntechV6Mapper {
...
+      return ResultStatus.PARTIAL
+    }
+
+    if (result.PendingTestCount === 0 && resultTotalCount > 0) {
+      return ResultStatus.COMPLETED
+    }
+
+    return status
+  }
+
```
Comment (monsieurBelbo) [inline src/providers/antechV6-mapper.ts:212]: Why move the method to another part of the file?
This makes reviewing the changes harder (everything is changed now) and it is unnecessary

File: src/providers/antechV6.mapper.spec.ts [hunk 3]
```diff
@@ -422,6 +439,192 @@ describe('AntechV6Mapper', () => {
...
+
+    it('should correctly map Tyroid Profile results items when first received', () => {
+      const tyroidProfileResult: AntechV6Result = tyroidResultsResponse_1[0]
+      const result: Result = mapper.mapAntechV6Result(tyroidProfileResult)
+      expect(result).toEqual(
+        expect.objectContaining({
+          id: '305580024',
+          orderId: '7092-VOY-37157652213',
+          accession: 'DLEA00533798',
+          status: ResultStatus.PARTIAL,
```
Comment (monsieurBelbo) [inline src/providers/antechV6.mapper.spec.ts:451]: How is this `PARTIAL`? No results are yet received.

File: src/providers/antechV6.mapper.spec.ts [hunk 4]
```diff
@@ -422,6 +439,192 @@ describe('AntechV6Mapper', () => {
...
+          testResults: expect.any(Array<TestResult>)
+        })
+      )
+      expect(result.testResults.length).toBe(4)
+      expect(result.testResults[0]).toEqual({
+        seq: 0,
+        code: 'SA380',
+        name: '',
+        items: expect.any(Array<TestResultItem>)
+      })
```
Comment (monsieurBelbo) [inline src/providers/antechV6.mapper.spec.ts:461]: Didn't we agree we would filter out final unit codes with no test results?

File: src/providers/antechV6.mapper.spec.ts [hunk 5]
```diff
@@ -640,4 +843,43 @@ describe('AntechV6Mapper', () => {
...
+      const result: AntechV6Result = { PendingTestCount: 2, TotalTestCount: 5 } as AntechV6Result
+      expect(mapper.extractResultStatus(result)).toBe(ResultStatus.PARTIAL)
+    })
+
+    it('should return COMPLETED if there are no pending tests', () => {
+      const result: AntechV6Result = { PendingTestCount: 0, TotalTestCount: 5 } as AntechV6Result
+      expect(mapper.extractResultStatus(result)).toBe(ResultStatus.COMPLETED)
+    })
+
+    it('should return PENDING if TotalTestCount is 0', () => {
```
Comment (monsieurBelbo) [inline src/providers/antechV6.mapper.spec.ts:869]: Should it?

File: src/providers/antechV6-mapper.ts (hunk 1 above)
Comment (monsieurBelbo) [reply src/providers/antechV6-mapper.ts]: I'm not sure if there are no tests to be run that it should be pending. Seems more like `COMPLETED` to me

**Comment by monsieurBelbo** [comment]
Still code committed with linting errors:
```
$ npx eslint "{src,apps,libs,test}/**/*.ts"

/Users/belbo/Nominal/dmi-engine-antech-v6-integration/src/providers/antechV6.mapper.spec.ts
  527:1  error  Delete `····`  prettier/prettier
  626:7  error  Delete `⏎`     prettier/prettier

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

File: src/providers/antechV6-mapper.ts (hunk 1 above)
Comment (monsieurBelbo) [reply src/providers/antechV6-mapper.ts:283]: OK, but there are no tests for the edge cases and the new changes implemented here. Please add them to the AntechV6Mapper spec.

**Comment by monsieurBelbo** [comment]
Please add tests for the new implementation of `AntechV6Mapper.extractResultStatus()`

Also, it is "thyroid" (with an "h")

**Review by monsieurBelbo — COMMENTED** [review]
Please use consistent values for `TotalTestCount` and `PendingTestCount` in the new tests.
